src/js/actions/layers.js

```diff
@@ -134,6 +132,135 @@ define(function (require, exports) {
      */
     var METADATA_NAMESPACE = global.EXTENSION_DATA_NAMESPACE;
 
+    var multiGetRef = function (docRef, layerRef, properties) {
+        return {
+            name: "get",
+            descriptor: {
+                "null": {
+                    "_multiGetRef": [
+                        {
+                            _propertyList: properties
+                        },
+                        layerRef,
+                        docRef
+                    ]
+                }
+                
+            },
+            "options": {
+                useMultiGet: true,
+                failOnMissingProperty: false
+            }
+        };
+    };
+    
+    var getLayerDescriptors = function (document, layers, options) {
+        layers = layers.toArray();
+        
+        if (layers.length === 0) {
```

Comment:
Alternate idea: can we just change batchPlay in `descriptor` to return a resolved promise when the list of commands is empty? Then we'd get this benefit everywhere without adding extra complexity to -design.